codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/HeaderField.java

```diff
@@ -72,9 +73,7 @@ public boolean equals(Object obj) {
             return false;
         }
         HeaderField other = (HeaderField) obj;
-        boolean nameEquals = HpackUtil.equals(name, other.name);
-        boolean valueEquals = HpackUtil.equals(value, other.value);
-        return nameEquals && valueEquals;
+        return (equalsConstantTime(name, other.name) & equalsConstantTime(value, other.value)) != 0;
```

Comment:
Are you using & to avoid short circuit behavior?  If so, that's subtle ans deserves a comment.   If not, then && seems appropriate.